fireant/queries/references.py

```diff
@@ -12,13 +12,16 @@ def adapt_for_reference_query(
     if reference_parts is None:
         return dimensions, metrics, filters
 
-    ref_dimension, time_unit, interval = reference_parts
+    ref_dim, unit, interval = reference_parts
+
+    trunc = database.trunc_date
+    offset_func = partial(database.date_add, date_part=unit, interval=interval)
+    offset_func_inv = partial(database.date_add, date_part=unit, interval=-interval)
+
+    ref_dimensions = _make_reference_dimensions(dimensions, ref_dim, offset_func, trunc)
```

Comment:
What is the reason the normal offset_func is used for the reference dimensions, but the reference filters use the inversed offset function?

Reply:
Nevermind, reading the tests code made this clearer

Reply:
yeah, ill leave a comment in a future commit.